### PR TITLE
OCM-5758 | fix: missing attributes from datasource of hcp mp

### PR DESCRIPTION
**Pull request by gdbranco**
[OCM-5758](https://issues.redhat.com//browse/OCM-5758)

**Change type**
- [ ] New feature
- [x] Bug fix
- [ ] Build
- [ ] CI
- [x] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [ ] Subsystem tests

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

**Comment by tphee** [comment]
I tried this datasource and getting nothing back.

Changing this line https://github.com/terraform-redhat/terraform-provider-rhcs/blob/main/provider/machinepool/hcp/machine_pool_resource.go#L467

`NodePool(state.ID.ValueString())`

to:

`NodePool(state.Name.ValueString())`

I am getting something back:
```
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "auto_repair": true,
            "autoscaling": {
              "enabled": false,
              "max_replicas": null,
              "min_replicas": null
            },
            "availability_zone": "us-east-1c",
            "aws_node_pool": {
              "instance_profile": "rosa-service-managed-production-29lu85bd0gcsdg4vqtdgfp4mqlsqjiks-hcp-worker",
              "instance_type": "m5.xlarge",
              "tags": null
            },
            "cluster": "29lu85bd0gcsdg4vqtdgfp4mqlsqjiks",
            "current_version": "4.14.8",
            "id": "workers-0",
            "labels": null,
            "name": "workers-0",
            "replicas": 1,
            "status": {
              "current_replicas": 1,
              "message": ""
            },
            "subnet_id": "subnet-038d7cfcb71e8b4ca",
            "taints": null,
            "tuning_configs": null,
            "upgrade_acknowledgements_for": null,
            "version": null
          },
          "sensitive_attributes": []
        }
      ]
```

**Comment by gdbranco** [comment]
@tphee 
> I tried this datasource and getting nothing back.

True, I'm more used to ID coming from API and CLI, but on terraform customer is usually not expected to supply it and instead it sets up this name. Let me change that

File: provider/machinepool/hcp/machine_pool_datasource.go [hunk 1]
```diff
@@ -156,6 +156,20 @@ func (r *HcpMachinePoolDatasource) Schema(ctx context.Context, req datasource.Sc
...
 			},
+			"version": schema.StringAttribute{
+				Description: "Desired version of OpenShift for the machine pool, for example '4.11.0'. If version is greater than the currently running version, an upgrade will be scheduled.",
+				Optional:    true,
+			},
+			"current_version": schema.StringAttribute{
+				Description: "The currently running version of OpenShift on the machine pool, for example '4.11.0'.",
+				Computed:    true,
+			},
+			"upgrade_acknowledgements_for": schema.StringAttribute{
```
[the remark this answers is not in the document]
Comment (nirarg) [reply provider/machinepool/hcp/machine_pool_datasource.go:167]: It was already done, ignore this :-)

**Review by ciaranRoche — COMMENTED** [review]
/lgtm

**Pull request merged**